src/backend/distributed/utils/maintenanced.c

```diff
@@ -309,35 +296,54 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		 * tasks should do their own time math about whether to re-run checks.
 		 */
 
-		if (secondsSincePrevStatsCollection >= STATISTICS_COLLECTION_INTERVAL ||
-			(prevStatsCollectionFailed &&
-			 secondsSincePrevStatsCollection >= STATISTICS_COLLECTION_RETRY_INTERVAL))
-		{
 #ifdef HAVE_LIBCURL
-			if (EnableStatisticsCollection)
-			{
-				MemoryContext statsCollectionContext =
-					AllocSetContextCreate(CurrentMemoryContext,
-										  "StatsCollection",
-										  ALLOCSET_DEFAULT_MINSIZE,
-										  ALLOCSET_DEFAULT_INITSIZE,
-										  ALLOCSET_DEFAULT_MAXSIZE);
-				MemoryContext oldContext =
-					MemoryContextSwitchTo(statsCollectionContext);
+		if (EnableStatisticsCollection &&
+			GetCurrentTimestamp() >= nextStatsCollectionTime)
+		{
+			bool statsCollectionSuccess = false;
+			InvalidateMetadataSystemCache();
+			StartTransactionCommand();
 
+			/*
+			 * Lock the extension such that it cannot be dropped or created
+			 * concurrently. Skip statistics collection if citus extension is
+			 * not accessible.
+			 *
+			 * Similarly, we skip statistics collection if there exists any
+			 * version mismatch or the extension is not fully created yet.
+			 */
+			if (!LockCitusExtension())
+			{
+				ereport(DEBUG1, (errmsg("could not lock the citus extension, "
```

Comment:
Shouldn't this be higher than `DEBUG1`? We'll never see it at that level. I could see `WARN` making sense.

Reply:
Oh, it's `DEBUG1` in deadlock detection? OK, then no problem leaving it the same here.

Reply:
Partly because we figured this might spam the logs in deadlock detection if a DROP EXTENSION is stuck. If you really want to see it you can set `log_min_messages` to DEBUG1 in postgresql.conf. I think we can just leave it as DEBUG1 in either case, since it's mostly interesting for debugging.